src/rules/quotemarkRule.ts

```diff
@@ -84,7 +84,9 @@ class QuotemarkWalker extends Lint.RuleWalker {
         const expectedQuoteMark = node.parent!.kind === ts.SyntaxKind.JsxAttribute ? this.jsxQuoteMark : this.quoteMark;
         const actualQuoteMark = node.getText()[0];
         if (actualQuoteMark !== expectedQuoteMark && !(this.avoidEscape && node.text.includes(expectedQuoteMark))) {
-            const escapedText = node.text.replace(new RegExp(expectedQuoteMark, "g"), `\\${expectedQuoteMark}`);
+            let escapedText = node.text.replace(new RegExp(expectedQuoteMark, "g"), `\\${expectedQuoteMark}`);
```

Comment:
try getting `node.getText()`, which will return a string with escaped characters, so you don't have to do the second replace. Otherwise you'd need to add a more replacement characters like "\\" and "\r".